Makefile

```diff
@@ -32,9 +33,8 @@ vet:
 	@$(GO) vet $(pkgs)
 
 build:
-	@echo ">> building binaries"
-	@./build/assets.sh
-	@./build/build.sh
+	@echo ">> building binaries using docker"
+	@docker run -i -v $(PWD):/go/src/github.com/google/cadvisor golang:$(goversion) /bin/sh -c "cd /go/src/github.com/google/cadvisor && go get github.com/tools/godep && ./build/assets.sh && ./build/build.sh"
```

Comment:
Given that we build entirely using vendor'ed code, I'm weary of mapping in
the entire `$GOPATH/pkg` directory. On top of that since we install godep,
it will overwrite the existing `godep` in the host's $GOBIN path, which is
undesirable.

On Tue, May 3, 2016 at 10:48 AM, Tim St. Clair <notifications@github.com>
wrote:

> In Makefile
> <https://github.com/google/cadvisor/pull/1251#discussion_r61924373>:
>
> > @@ -32,9 +33,8 @@ vet:
> >  	@$(GO) vet $(pkgs)
> >
> >  build:
> > -	@echo ">> building binaries"
> > -	@./build/assets.sh
> > -	@./build/build.sh
> > +	@echo ">> building binaries using docker"
> > +	@docker run -i -v $(PWD):/go/src/github.com/google/cadvisor golang:$(goversion) /bin/sh -c "cd /go/src/github.com/google/cadvisor && go get github.com/tools/godep && ./build/assets.sh && ./build/build.sh"
>
> I don't think it is. I think you need to mount the $GOPATH/pkg directory
> for it to work.
>
> With your changes (irrelevant output omitted):
>
> [(c262eb7...)] cadvisor: time make build
> >> building binaries using docker
>  >   cadvisor
>
> real    0m14.201s
> user    0m0.066s
> sys 0m0.032s
>
> [(c262eb7...)] cadvisor: time make build
> >> building binaries using docker
>  >   cadvisor
>
> real    0m14.279s
> user    0m0.073s
> sys 0m0.027s
>
> Without your changes:
>
> [install] cadvisor: time make build
> >> building binaries
>  >   cadvisor
>
> real    0m6.658s
> user    0m13.909s
> sys 0m1.622s
>
> [install] cadvisor: time make build
> >> building binaries
>  >   cadvisor
>
> real    0m0.781s
> user    0m0.343s
> sys 0m0.188s
>
> Notice how the second build is super fast once everything is cached.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/google/cadvisor/pull/1251/files/97ac208588777d15872e39ffe615760b853c91c4#r61924373>
>